src/hooks/useDragAndDrop.ts

```diff
@@ -75,6 +83,7 @@ export default function useDragAndDrop({dropZone, onDrop = () => {}, shouldAllow
                 case DRAG_ENTER_EVENT:
                     dragCounter.current++;
                     setDropEffect(event);
+                    onDragEnter(event);
```

Comment:
Could we call the `PopoverContext` `close` method here?

Reply:
`PopoverContext` is `App`-level and it's not likely to change. Still, a conditional call as a defense-in-depth might be considered.

While it adds some coupling, it's an acceptable level in my opinion.

This would remove some duplication and could simplify the overall solution if I don't miss some unexpected consequences of this change.